pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -756,14 +756,58 @@ func (s *FlinkStateMachine) handleSubmittingJob(ctx context.Context, app *v1beta
 		return statusUnchanged, errors.Errorf("Could not find job %s", s.flinkController.GetLatestJobID(ctx, app))
 	}
 
-	// wait until all vertices have been scheduled and started
-	allVerticesStarted := true
-	for _, v := range job.Vertices {
-		allVerticesStarted = allVerticesStarted && (v.StartTime > 0)
+	jobStartTime := getJobStartTimeInUTC(job.StartTime)
+	now := time.Now().UTC()
+	cfg := config.GetConfig()
+	logger.Info(ctx, "Job vertex timeout config is ", cfg.FlinkJobVertexTimeout)
+	flinkJobVertexTimeout := cfg.FlinkJobVertexTimeout
+	if now.Before(jobStartTime.Add(flinkJobVertexTimeout.Duration)) {
+		allVerticesRunning, hasFailure, failedVertexIndex := monitorAllVerticesState(job)
+
+		// fail fast
+		if hasFailure {
+			s.flinkController.LogEvent(ctx, app, corev1.EventTypeWarning, "JobRunningFailed",
+				fmt.Sprintf(
+					"Vertex %d with name [%s] state is Failed", failedVertexIndex, job.Vertices[failedVertexIndex].Name))
```

Comment:
Curious, will this log message make sense to users?

Reply:
An example of the message is like below. I think it gives hints to point out which vertex may has problem

```
Vertex 2 with name [[9]{Process events with stateful dofn, nmodes_filter_15m_data, filter_active_sessions, nmodes_timestamp, __aggregate_event_processing_results_timestamp, nmodes_window_15m, __aggregate_event_processing_results_filter_up5m_data, __aggregate_event_processing_results_window_1m, __aggregate_event_processing_results_window_5m} -> ToKeyedWorkItem] state is Failed
```

Reply:
Thanks!